CMakeLists.txt

```diff
@@ -206,6 +206,7 @@ check_function_exists(pthread_attr_setcpupercent_np HAVE_PTHREAD_ATTR_SETCPUPERC
 check_function_exists(pthread_yield_np HAVE_PTHREAD_YIELD_NP)
 check_function_exists(pthread_main_np HAVE_PTHREAD_MAIN_NP)
 check_function_exists(pthread_workqueue_setdispatch_np HAVE_PTHREAD_WORKQUEUE_SETDISPATCH_NP)
+check_function_exists(pthread_setname_np HAVE_PTHREAD_SETNAME_NP)
```

Comment:
Some platforms have similar functionality spelled `pthread_set_name_np`. Is the actual logic  potentially applicable to other platforms, or is it Linux-specific?

Reply:
>  Is the actual logic potentially applicable to other platforms, or is it Linux-specific?

`pthread_set_name_np` and `pthread_setname_np` have a different signature on OpenBSD and FreeBSD.

From the FreeBSD man page
```
void pthread_set_name_np(pthread_t thread, const char	*name);

int pthread_setname_np(pthread_t thread, const char *name);
```

Platforms supporting `pthread_setname_np` share the same signature.

Reply:
Sure, but if we check for pthread_set_name_np as well as pthread_setname_np, can we make the code you added in src/queue.c work for other platforms, and not just Linux? Or is your code Linux-specific?

I would check myself, but I can't for another week or so.